src/twisted/web/wsgi.py

```diff
@@ -506,11 +507,13 @@ def run(self):
         except:
             def wsgiError(started, type, value, traceback):
                 err(Failure(value, type, traceback), "WSGI application error")
-                if started:
-                    self.request.loseConnection()
-                else:
-                    self.request.setResponseCode(INTERNAL_SERVER_ERROR)
-                    self.request.finish()
+                if not self._requestFinished:
```

Comment:
minor comment,
to reduce indentation and document the intend maybe this can be written as
```
def wsgiError(started, type, value, traceback):
    err(Failure(value, type, traceback), "WSGI application error")
    if self._requestFinished:
        # Some comment about what happend here
        return
    if started:
      EXISTING_CODE_HERE
```